lib/elastic_whenever/option.rb

```diff
@@ -133,10 +133,11 @@ def initialize(args)
                      elsif access_key && secret_key
                        Aws::Credentials.new(access_key, secret_key)
                      end
+      @credentials = nil unless @credentials&.set?
```

Comment:
just a little explainer: when using SSO credentials, the `Aws::SharedCredentials` instance's `credentials` will not be set! that's why I chose to `nil` the `@credentials` again in that case.

Reply:
Considering that `profile` is not just for shared credentials, I think it is probably not appropriate to return `Aws::SharedCredentials` when the `profile` exists.
It might make more sense if including the `profile` in `aws_config` means you don't have to explicitly return `SharedCredentials` here.

e.g.

```ruby
@credentials = if access_key && secret_key
                 Aws::Credentials.new(access_key, secret_key)
               end
```

Reply:
Yeah, I agree that this approach is valid. However, since there is no need to explicitly return `SharedCredentials` in the first place, I'm concerned that when looking at it later, it may be difficult to understand why `SharedCredentials` is returned here. At least I couldn't understand why this line was needed without explanation.

It is possible to merge this change, but I will probably release it after fixing it as above.

Reply:
Okay question is how you would find out that SharedCredentials aren't needed in the first place without adding a lot more code. If you feel like adding more code for solving the same problem, it's up to you. Happy if you just take it up then. Meanwhile we can keep using our fork then.

Reply:
Maybe I got you wrong, so you propose to just remove the shared credentials completely? I can try and patch up the tests.

Reply:
I applied the change, I think it's valid to remove the SharedCredentials completely since AWS sdk should handle the credentials if none are passed via a key pair.

Reply:
Not to mention that passing in an AWS key pair via CLI options is a security risk as credentials will pretty likely end up in history or logs. But that's not my concern here :)